include/nbl/asset/ICPUMeshBuffer.h

```diff
@@ -57,580 +58,748 @@ namespace impl
     }
 }
 
-class ICPUMeshBuffer : public IMeshBuffer<ICPUBuffer, ICPUDescriptorSet, ICPURenderpassIndependentPipeline>, public BlobSerializable, public IAsset
+class ICPUMeshBuffer final : public IMeshBuffer<ICPUBuffer,ICPUDescriptorSet,ICPURenderpassIndependentPipeline,ICPUSkeleton>, public BlobSerializable, public IAsset
 {
-    using base_t = IMeshBuffer<ICPUBuffer, ICPUDescriptorSet, ICPURenderpassIndependentPipeline>;
-    //vertices
-    uint32_t posAttrId = 0u;
-    uint32_t normalAttrId = MAX_VERTEX_ATTRIB_COUNT;
+        using base_t = IMeshBuffer<ICPUBuffer,ICPUDescriptorSet,ICPURenderpassIndependentPipeline,ICPUSkeleton>;
+        // knowing the position attribute ID is important for AABB computations etc.
+        uint32_t posAttrId : 5;
+        uint32_t normalAttrId : 5;
+        // by having one attribute only, we limit the number of bones per vertex to 4
+        uint32_t jointIDAttrId : 5;
+        uint32_t jointWeightAttrId : 5;
+
+    protected:
+        virtual ~ICPUMeshBuffer() = default;
+
+    public:
+        //! Default constructor (initializes pipeline, desc set and buffer bindings to nullptr)
+        ICPUMeshBuffer() : base_t(nullptr, nullptr, nullptr, SBufferBinding<ICPUBuffer>{})
+        {
+            posAttrId = 0u;
+            normalAttrId = MAX_VERTEX_ATTRIB_COUNT;
+            jointIDAttrId = MAX_VERTEX_ATTRIB_COUNT;
+            jointWeightAttrId = MAX_VERTEX_ATTRIB_COUNT;
+        }
+        using base_t::base_t;
```

Comment:
with inherited constructors `*AttrId` members wont be initialized

Reply:
good catch